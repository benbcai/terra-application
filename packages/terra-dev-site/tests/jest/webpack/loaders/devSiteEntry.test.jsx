const path = require('path');
const { runLoaders } = require('loader-runner');

describe('devSiteEntry', () => {
  it('It sets up the query param', (done) => {
    runLoaders({
      resource: `${path.resolve(__dirname, '..', '..', '..', '..', 'src', 'webpack', 'templates', 'entry.template')}?query`,
      loaders: [
        {
          loader: path.resolve(__dirname, '../../../../src/webpack/loaders/devSiteEntry'),
          options: {
            entryMapping: {
              '?query': '?bootstrapQuery',
            },
          },
        },
      ],
    }, (err, result) => {
      if (err) return done(err);
      expect(result.result[0]).toMatchSnapshot();
      return done();
    });
  });
});
