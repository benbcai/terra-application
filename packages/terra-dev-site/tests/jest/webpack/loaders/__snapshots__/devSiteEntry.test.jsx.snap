// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`devSiteEntry It sets up the query param 1`] = `
"import('./bootstrap.template?bootstrapQuery');
"
`;
